src/_posts/databases/postgresql/2000-01-01-upgrading.md

```diff
@@ -0,0 +1,233 @@
+
+We usually roughly estimate the downtime caused by the operation to a few
+seconds (2 to 10 seconds), depending on the platform load.
+
+#### For Business Plans
+
+1. The standby instance is stopped. The primary instance is still running, so
+   the database is still reachable.
+2. We restart the standby instance with the targeted version.
+3. When the standby instance is ready, a failover is done to make it primary.
+   The old primary becomes the standby instance. During this operation, the
+   connection can be lost during a few milliseconds. The database is still
+   reachable.
+4. The new standby instance is restarted with the targeted version.
+5. Once restarted, the cluster is fully upgraded and fully operational again.
+6. The application is restarted to ensure proper connexions. This shouldn't
+   cause any additional downtime.
+
+Minor-upgrades of Business plans are **usually achieved without any downtime**.
+Note that a few milliseconds downtime can still occur during the failover.
+
+
+## Understanding the Major-Upgrade Process
+
+### Prerequisites
+
+To get access to the next major version, your database must first be upgraded
+to the latest minor version. Each following major version will be suggested
+until you reach the latest version available on the platform.
+
+Let’s assume your database is currently on PostgreSQL® 13.5 and you want to
+upgrade to PostgreSQL® 15. First, upgrade your database to the latest version
+of the 13.x branch. Then, proceed to the latest version of the 14.x branch.
+Finally, upgrade to the latest version of the 15.x branch.
+
+### Process
+
+#### For Starter Plans
+
+1. The instance is stopped. The database is unreachable.
+2. A new primary instance is booted with the targeted version.
+3. `pg_upgrade` is executed on this new instance.
+4. The instance is restarted. The database is reachable again and the
+   application can use it normally.
+5. The `ANALYZE` SQL command is executed against the database to build up
+   PostgreSQL® statistics. PostgreSQL® uses these statistics to determine
+   the most efficient execution plans for queries.
+6. The application is restarted to ensure proper connexions. This shouldn't
+   cause any additional downtime.
+7. A base backup is asynchronously done to make [point-in-time recovery]({% post_url databases/postgresql/2000-01-01-backing-up %}#understanding-point-in-time-recovery-backups)
+   available again.
+
+Since we have to completely stop the instance to upgrade it, **a downtime is
+inevitable**.
+
+We usually roughly estimate the overall downtime of the operation by assuming
+1 minute of unavailibility per 10GiB of data. This remains a raw estimation and
+our experience tends to show that it often takes less time.
+
+#### For Business Plans
+
+1. The entire cluster is stopped. The database is unreachable.
+2. A new primary instance is booted with the targeted version.
+3. `pg_upgrade` is executed on this new primary instance.
```

Comment:
issue: I don't think step 2 is correct. From what I remember `pg_upgrade` is executed on the data and we don't need any instance to be booted.